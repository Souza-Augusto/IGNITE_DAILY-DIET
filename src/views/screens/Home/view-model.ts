import { useState } from 'react';
import { GetMeals } from '@storage/get-meals';
import { useTheme } from 'styled-components/native';
import { CountMeals } from '@utils/meals/count-meals';
import { CountHealthyMeals } from '@utils/meals/count-healthy-meals';
import { SeparateByDate } from '@utils/meals/separate-by-date';
import { mealDTO } from '@dtos/meal-dtos';
import { useNavigation } from '@react-navigation/native';
import { CountOffDietMeals } from '@utils/meals/count-off-diet-meals';

type sectionListDataProps = {
  title: string;
  data: mealDTO[];
};

interface HomeViewModelProps {
  data: sectionListDataProps[];
  loading: boolean;
  fetchMeals: () => void;
  dialogVisible: boolean;
  setDialogVisible: React.Dispatch<React.SetStateAction<boolean>>;
  dialogTitle: string;
  setDialogTitle: React.Dispatch<React.SetStateAction<string>>;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
  handleNavigateStatistcs: () => void;
  handleNavigateMealDetails: (item: mealDTO) => void;
  handleNavigateResgisterMeal: () => void;
  cardColor: string;
  arrowIcon: string;
  percentage: string;
}

function useHomeViewModel(): HomeViewModelProps {
  const [data, setData] = useState<sectionListDataProps[]>([]);
  const [loading, setLoading] = useState(true);
  const [dialogVisible, setDialogVisible] = useState(false);
  const [dialogTitle, setDialogTitle] = useState('');
  const [percentage, setPercentage] = useState('');
  const [cardColor, setCardColor] = useState('');
  const [arrowIcon, setArrowIcon] = useState('');

  const { navigate } = useNavigation();

  const { COLORS } = useTheme();

  function calculatePercentage(data: sectionListDataProps[]) {
    const meals = CountMeals(data);
    const healthy = CountHealthyMeals(data);
    const offDiet = CountOffDietMeals(data);

    const healthypercentage = (healthy / meals) * 100;

    if (meals === 0) {
      setPercentage('0,00%');
    } else {
      setPercentage(
        String(healthypercentage.toFixed(2) + '%').replace('.', ',')
      );
    }

    if (offDiet < healthy) {
      setCardColor(COLORS.GREEN_LIGHT);
      setArrowIcon(COLORS.GREEN_DARK);
      return;
    }
    if (offDiet > healthy) {
      setCardColor(COLORS.RED_LIGHT);
      setArrowIcon(COLORS.RED_DARK);
      return;
    }
    setCardColor(COLORS.GRAY_600);
    setArrowIcon(COLORS.GRAY_200);
  }

  async function fetchMeals() {
    try {
      setLoading(true);
      const meals = await GetMeals();

      const separateByDates = SeparateByDate(meals);

      calculatePercentage(separateByDates);

      setData(separateByDates);

      setLoading(false);
    } catch (error) {
      setLoading(false);
      setDialogTitle('Não foi possível carregar as refeições.');
      setDialogVisible(true);
    }
  }

  function handleNavigateStatistcs() {
    navigate('statistics', { meals: data });
  }

  function handleNavigateMealDetails(item: mealDTO) {
    navigate('meal-details', { meal: item });
  }

  function handleNavigateResgisterMeal() {
    navigate('register-meal');
  }

  return {
    data,
    fetchMeals,
    loading,
    dialogTitle,
    dialogVisible,
    setDialogTitle,
    setDialogVisible,
    setLoading,
    handleNavigateMealDetails,
    handleNavigateResgisterMeal,
    handleNavigateStatistcs,
    arrowIcon,
    cardColor,
    percentage,
  };
}
export { useHomeViewModel };
