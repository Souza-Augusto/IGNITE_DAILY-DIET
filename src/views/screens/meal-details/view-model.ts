import { useState, Dispatch } from 'react';
import { useRoute, useNavigation } from '@react-navigation/native';
import { mealDTO } from 'src/dtos/mealDTO';
import { DeleteMeal } from '@storage/deleteMeal';

type RouteParams = {
  meal: mealDTO;
};

interface MealDetailsProps {
  handleGoBack: () => void;
  dialogVisible: boolean;
  dialogMessage: string;
  meal: mealDTO;
  alert: () => void;
  deleteMeal: () => void;
  setDialogVisible: Dispatch<React.SetStateAction<boolean>>;
  handleNavigationRegisterMeal: () => void;
  dialogPositiveButtonTitle: string | null;
}

function useMealDetailsViewModel(): MealDetailsProps {
  const [dialogVisible, setDialogVisible] = useState(false);
  const [dialogMessage, setDialogMessage] = useState('');
  const [dialogPositiveButtonTitle, setDialogdialogPositiveButtonTitle] =
    useState<string | null>('Sim, excluir');

  const navigation = useNavigation();

  const route = useRoute();
  const { meal } = route.params as RouteParams;

  function alert() {
    setDialogdialogPositiveButtonTitle('Sim, excluir');
    setDialogMessage('Deseja realmente excluir o registro da refeição?');
    setDialogVisible(true);
  }

  async function deleteMeal() {
    setDialogVisible(false);
    try {
      await DeleteMeal(meal);
      navigation.navigate('home');
    } catch (error) {
      console.log(error);
      setDialogMessage('Não foi possível excluir a refeição.');
      setDialogdialogPositiveButtonTitle(null);
      setDialogVisible(true);
    }
  }

  function handleGoBack() {
    navigation.goBack();
  }

  function handleNavigationRegisterMeal() {
    navigation.navigate('register-meal', { meal });
  }

  return {
    alert,
    deleteMeal,
    dialogVisible,
    meal,
    handleGoBack,
    dialogMessage,
    setDialogVisible,
    handleNavigationRegisterMeal,
    dialogPositiveButtonTitle,
  };
}

export { useMealDetailsViewModel };
