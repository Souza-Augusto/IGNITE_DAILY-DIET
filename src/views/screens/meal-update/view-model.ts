import { useNavigation, useRoute } from '@react-navigation/native';
import { mealDTO } from '@dtos/meal-dto';
import { updateMeal } from '@storage/update-meal';
import { useState } from 'react';

type RouteParams = {
  meal: mealDTO;
};

export interface MealUpdateProps {
  healthy: boolean;
  setHealthy: React.Dispatch<React.SetStateAction<boolean>>;
  name: string;
  setName: React.Dispatch<React.SetStateAction<string>>;
  description: string;
  setDescription: React.Dispatch<React.SetStateAction<string>>;
  date: string;
  setDate: React.Dispatch<React.SetStateAction<string>>;
  hour: string;
  setHour: React.Dispatch<React.SetStateAction<string>>;
  isLoading: boolean;
  dialogVisible: boolean;
  setDialogVisible: React.Dispatch<React.SetStateAction<boolean>>;
  dialogMessage: string;
  handleMealUpdate: () => void;
  fetchData: () => void;
}

function useMealUpdateViewModel(): MealUpdateProps {
  const [healthy, setHealthy] = useState<boolean | null>(null);
  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const [date, setDate] = useState('');
  const [hour, setHour] = useState('');
  const [isLoading, setIsLoading] = useState(false);
  const [dialogVisible, setDialogVisible] = useState(false);
  const [dialogMessage, setDialogMessage] = useState('');

  const currentDate = new Date();

  const { navigate } = useNavigation();

  const route = useRoute();
  const params = route.params as RouteParams;

  function fetchData() {
    setName(params?.meal.name);
    setDescription(params?.meal.description);
    setDate(params?.meal.date);
    setHour(params?.meal.hour);
    setHealthy(params?.meal.healthy);
  }

  async function handleMealUpdate() {
    try {
      if (
        name.trim().length <= 0 ||
        description.trim().length <= 0 ||
        date.trim().length <= 0 ||
        hour.trim().length <= 0
      ) {
        setDialogMessage('Preencha todos os campos.');
        setDialogVisible(true);
        return;
      }

      const [day, month, year] = date.split('/');

      const parsedDay = parseInt(day, 10);
      const parsedMonth = parseInt(month, 10);
      const parsedYear = parseInt(year, 10);

      if (
        date.length < 10 ||
        parsedDay < 1 ||
        parsedDay > 31 ||
        parsedMonth < 1 ||
        parsedMonth > 12 ||
        parsedYear < 2023
      ) {
        setDialogMessage('Formato de data inválido.');
        setDialogVisible(true);
        return;
      }

      const [hours, minutes] = hour.split(':');
      const parsedHours = parseInt(hours, 10);
      const parsedMinutes = parseInt(minutes, 10);

      if (
        hour.length < 5 ||
        parsedHours < 0 ||
        parsedHours > 23 ||
        parsedMinutes < 0 ||
        parsedMinutes > 59
      ) {
        setDialogMessage('Hora inválida.');
        setDialogVisible(true);
        return;
      }
      if (healthy === null) {
        setDialogMessage(
          'Por favor informe se a refeição que você deseja está dentro da dieta.'
        );
        setDialogVisible(true);
        return;
      }
      setIsLoading(true);

      await updateMeal({
        id: params?.meal.id,
        hour,
        date,
        name,
        healthy: healthy as boolean,
        description,
        createdAt: params?.meal.createdAt,
        updatedAt: String(currentDate),
      });
      setIsLoading(false);
      navigate('status-noticed', { healthy: healthy as boolean });
    } catch (error) {
      console.log(error);
      setDialogMessage(`Não foi possível alterar a refeição`);
      setDialogVisible(true);
    }
  }

  return {
    date,
    description,
    dialogMessage,
    dialogVisible,
    setDialogVisible,
    healthy: healthy as boolean,
    hour,
    isLoading,
    name,
    setDate,
    setDescription,
    setHealthy: setHealthy as React.Dispatch<React.SetStateAction<boolean>>,
    setHour,
    setName,
    handleMealUpdate,
    fetchData,
  };
}

export { useMealUpdateViewModel };
