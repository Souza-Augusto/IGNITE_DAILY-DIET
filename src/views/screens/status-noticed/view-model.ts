import { useRoute, useNavigation } from '@react-navigation/native';

type RouteParams = {
  healthy: boolean;
};

export interface StatusNoticeProps {
  healthy: boolean;
  handleNavigateHome: () => void;
}

function useStatusNoticedViewModel(): StatusNoticeProps {
  const route = useRoute();
  const { healthy } = route.params as RouteParams;
  const { navigate } = useNavigation();

  function handleNavigateHome() {
    navigate('home');
  }

  return { healthy, handleNavigateHome };
}
export { useStatusNoticedViewModel };
