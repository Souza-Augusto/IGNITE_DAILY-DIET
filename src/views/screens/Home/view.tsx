import { useCallback } from 'react';
import { Header } from '@components/Header/view';
import {
  Card,
  CardDescription,
  CardTitle,
  Container,
  ArrowUpRightIcon,
  Title,
  Date,
  DateContainer,
  PlusIcon,
} from './styles';

import { SectionList } from 'react-native';

import { Button } from '@components/Button/view';
import { Meal_Card } from '@components/Meal_Card/view';
import { Loading } from '@components/Loading/view';

import { useFocusEffect, useNavigation } from '@react-navigation/native';

import { ListEmpty } from '@components/ListEmpty/view';

import { useHomeViewModel } from './view-model';
import { Dialog } from '@components/dialog/view';

export function Home() {
  const {
    countMeals,
    data,
    fetchMeals,
    healthyMeals,
    loading,
    dialogTitle,
    dialogVisible,
    setDialogTitle,
    setDialogVisible,
    setLoading,
  } = useHomeViewModel();

  useFocusEffect(
    useCallback(() => {
      try {
        fetchMeals();
      } catch (error) {
        setLoading(false);
        setDialogTitle('Não foi possível carregar as refeições.');
        setDialogVisible(true);
      }
    }, [])
  );

  const navigation = useNavigation();
  if (loading) {
    return <Loading />;
  }
  return (
    <Container>
      <Dialog
        negativeFunction={() => setDialogVisible(false)}
        dialogVisible={dialogVisible}
        dialogMessage={dialogTitle}
      />
      <Header />
      <Card
        percentege={(healthyMeals / countMeals) * 100}
        activeOpacity={0.5}
        onPress={() => navigation.navigate('statistics', { meals: data })}
      >
        <ArrowUpRightIcon
          name='arrow-up-right'
          size={25}
          percentege={(healthyMeals / countMeals) * 100}
        />
        <CardTitle>
          {countMeals === 0
            ? '0,00%'
            : String(
                ((healthyMeals / countMeals) * 100).toFixed(2) + '%'
              ).replace('.', ',')}
        </CardTitle>
        <CardDescription>das refeições dentro da sua dieta</CardDescription>
      </Card>
      <Title>Refeições</Title>
      <Button
        onPress={() => navigation.navigate('registerMeal')}
        title='Nova refeição'
        before={<PlusIcon />}
      />
      <SectionList
        contentContainerStyle={data?.length === 0 && { flex: 1 }}
        showsVerticalScrollIndicator={false}
        sections={data}
        keyExtractor={(_, index) => index.toString()}
        renderSectionHeader={({ section: { title } }) => (
          <DateContainer>
            <Date>{title}</Date>
          </DateContainer>
        )}
        renderItem={({ item }) => (
          <Meal_Card
            onPress={() =>
              navigation.navigate('mealDetails', {
                meal: item,
              })
            }
            hour={item.hour}
            meal={item.name}
            healthy={item.healthy}
          />
        )}
        ListEmptyComponent={
          <ListEmpty message='Sua lista de refeições está vazia, que tal cadastrar uma nova refeição?' />
        }
      />
    </Container>
  );
}
