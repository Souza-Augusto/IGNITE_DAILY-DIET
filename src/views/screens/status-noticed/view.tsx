import {
  ConfirmRegister,
  FeaturedNoticed,
  Image,
  Container,
  Title,
  Noticed,
} from './styles';

import { useStatusNoticedViewModel } from './view-model';

export function StatusNoticed() {
  const { handleNavigateHome, healthy } = useStatusNoticedViewModel();

  return (
    <Container testID='status-noticed-container'>
      <Title mealType={healthy}>
        {healthy ? 'Continue assim!' : 'Que pena!'}
      </Title>
      <Noticed>
        {healthy ? (
          <>
            Você continua <FeaturedNoticed>dentro da dieta.</FeaturedNoticed>{' '}
            Muito bem!
          </>
        ) : (
          <>
            Você <FeaturedNoticed>saiu da dieta</FeaturedNoticed> dessa vez, mas
            continue se esforçando e não desista!
          </>
        )}
      </Noticed>
      <Image
        testID='image'
        source={
          healthy
            ? require('@assets/images/png/Illustration(1).png')
            : require('@assets/images/png/Illustration(2).png')
        }
      />
      <ConfirmRegister
        onPress={handleNavigateHome}
        title='Ir para a tela inicial'
      />
    </Container>
  );
}
