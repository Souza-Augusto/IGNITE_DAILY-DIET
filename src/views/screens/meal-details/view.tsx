import {
  Container,
  DetaisMealContainer,
  Meal,
  Description,
  DateTimeTitle,
  DateTime,
  MealTypeContainer,
  MealType,
  MealTypeDescription,
  ButtonContainer,
  Button,
  PencilIcon,
} from './styles';
import Trash from '@assets/images/svg/Trash.svg';
import { Dialog } from '@components/dialog/view';
import { useMealDetailsViewModel } from './view-model';
import { Header } from '@components/Header';

export function MealDetails() {
  const {
    alert,
    deleteMeal,
    dialogMessage,
    dialogVisible,
    meal,
    handleNavigationRegisterMeal,
    handleGoBack,
    setDialogVisible,
    dialogPositiveButtonTitle,
  } = useMealDetailsViewModel();

  return (
    <Container mealType={meal.healthy}>
      <Dialog
        dialogVisible={dialogVisible}
        dialogMessage={dialogMessage}
        positiveFunction={deleteMeal}
        negativeFunction={() => setDialogVisible(false)}
        positiveButtonTitle={dialogPositiveButtonTitle}
        negativeButtonTitle='Não'
      />

      <Header.Root>
        <Header.BackButton />
        <Header.Title title={'Refeição'} />
      </Header.Root>
      <DetaisMealContainer>
        <Meal>{meal.name}</Meal>
        <Description>{meal.description}</Description>
        <DateTimeTitle>Data e Hora</DateTimeTitle>
        <DateTime>
          {meal.date} às {meal.hour}
        </DateTime>
        <MealTypeContainer>
          <MealType mealType={meal.healthy} />
          <MealTypeDescription numberOfLines={1}>
            {meal.healthy ? 'dentro da dieta' : 'fora da dieta'}
          </MealTypeDescription>
        </MealTypeContainer>
        <ButtonContainer>
          <Button
            onPress={handleNavigationRegisterMeal}
            before={<PencilIcon />}
            title='Editar Refeição'
          />
          <Button
            before={<Trash />}
            onPress={alert}
            type='SECONDARY'
            title='Excluir Refeição'
          />
        </ButtonContainer>
      </DetaisMealContainer>
    </Container>
  );
}
