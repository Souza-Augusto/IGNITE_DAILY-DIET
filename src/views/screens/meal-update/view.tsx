import { useEffect } from 'react';
import { Header } from '@components/Header';
import {
  Container,
  InputTitle,
  RegisterMealContainer,
  InputDescription,
  DateTimeInputContainer,
  InputMaskContainer,
  DateTime,
  Question,
  MealsTypeButtonContainer,
  MealsType,
  DateTimeInput,
  OffDietButton,
  HealthyMealButton,
} from './styles';
import { Button } from '@components/Button/view';
import { Input } from '@components/Input/view';
import { ScrollView } from 'react-native';

import { Loading } from '@components/Loading/view';
import { useMealUpdateViewModel } from './view-model';
import { Dialog } from '@components/dialog/view';

export function MealUpdate() {
  const {
    date,
    description,
    dialogMessage,
    dialogVisible,
    setDialogVisible,
    healthy,
    hour,
    isLoading,
    name,
    setDate,
    setDescription,
    setHealthy,
    setHour,
    setName,
    handleMealUpdate,
    fetchData,
  } = useMealUpdateViewModel();

  useEffect(() => {
    fetchData();
  }, []);
  if (isLoading) {
    <Loading />;
  }
  return (
    <Container>
      <Dialog
        negativeButtonTitle='Fechar'
        negativeFunction={() => setDialogVisible(false)}
        dialogMessage={dialogMessage}
        dialogVisible={dialogVisible}
      />
      <Header.Root>
        <Header.BackButton />
        <Header.Title title='Alterar Refeição' />
      </Header.Root>

      <RegisterMealContainer>
        <ScrollView showsVerticalScrollIndicator={false}>
          <InputTitle>Nome</InputTitle>

          <Input
            placeholder='Ex: Saduíche'
            value={name}
            onChangeText={setName}
          />

          <InputTitle>Descrição</InputTitle>

          <InputDescription
            placeholder='Ex: Sanduíche de pão integral com atum e salada de alface e tomate.'
            value={description}
            multiline
            onChangeText={setDescription}
          />

          <DateTimeInputContainer>
            <InputMaskContainer>
              <DateTime>Data</DateTime>

              <DateTimeInput
                placeholder='DD/MM/YYYY'
                value={date}
                onChangeText={setDate}
                type='datetime'
                options={{
                  format: 'DD/MM/YYYY',
                }}
              />
            </InputMaskContainer>
            <InputMaskContainer>
              <DateTime>Hora</DateTime>

              <DateTimeInput
                placeholder='00:00'
                value={hour}
                onChangeText={setHour}
                type='datetime'
                options={{
                  format: 'HH:mm',
                }}
              />
            </InputMaskContainer>
          </DateTimeInputContainer>
          <Question>Está dentro da dieta?</Question>
          <MealsTypeButtonContainer>
            <HealthyMealButton
              onPress={() => setHealthy(true)}
              mealType={healthy}
              type='SECONDARY'
              title='SIM'
              before={<MealsType mealType />}
            />
            <OffDietButton
              type='SECONDARY'
              mealType={healthy}
              onPress={() => setHealthy(false)}
              title='NÃO'
              before={<MealsType mealType={false} />}
            />
          </MealsTypeButtonContainer>
        </ScrollView>

        <Button
          style={{ marginBottom: 10 }}
          onPress={handleMealUpdate}
          title='Salvar alterações'
        />
      </RegisterMealContainer>
    </Container>
  );
}
