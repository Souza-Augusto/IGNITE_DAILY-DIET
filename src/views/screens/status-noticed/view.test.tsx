import { fireEvent, render } from '@__tests__/utils/customRenderItem';
import { StatusNoticeProps, useStatusNoticedViewModel } from './view-model';
import { StatusNoticed } from './view';
import '@testing-library/react-native/extend-expect';

interface CreateMockProps {
  healthy: boolean;
}

interface factoryProps {
  healthy: boolean;
}

jest.mock('./view-model');
jest.mock('@assets/images/png/Illustration(1).png', () => 1);
jest.mock('@assets/images/png/Illustration(2).png', () => 2);

beforeEach(() => jest.clearAllMocks());

const mockHandleNavigateHome = jest.fn();
const createMockViewmodel = ({
  healthy,
}: CreateMockProps): StatusNoticeProps => ({
  healthy,
  handleNavigateHome: mockHandleNavigateHome,
});

function makeSut({ healthy }: factoryProps) {
  const MOCK_VIEW_MODEL = createMockViewmodel({
    healthy,
  });

  jest.mocked(useStatusNoticedViewModel).mockReturnValueOnce(MOCK_VIEW_MODEL);

  const CONTAINER = render(<StatusNoticed />);

  return CONTAINER;
}

describe('Screen:status-noticed view-model', () => {
  it('Should be render correctly', () => {
    const { toJSON, getByTestId } = makeSut({ healthy: true });

    const CONTAINER = getByTestId('status-noticed-container');

    expect(CONTAINER).toBeVisible();

    expect(toJSON).toMatchSnapshot();
  });
  it('Should be title render correctly when healthy is true', () => {
    const { toJSON, getByText } = makeSut({ healthy: true });

    const TITLE = getByText('Continue assim!');

    expect(TITLE).toBeVisible();

    expect(toJSON).toMatchSnapshot();
  });

  it('Should be title render correctly when healthy is false', () => {
    const { toJSON, getByText } = makeSut({ healthy: false });

    const TITLE = getByText('Que pena!');

    expect(TITLE).toBeVisible();

    expect(toJSON).toMatchSnapshot();
  });
  it('Should be Noticed render correctly when healthy is true', () => {
    const { toJSON, getByText } = makeSut({ healthy: true });

    const TITLE = getByText('Você continua dentro da dieta. Muito bem!');

    expect(TITLE).toBeVisible();

    expect(toJSON).toMatchSnapshot();
  });
  it('Should be Noticed render correctly when healthy is false', () => {
    const { toJSON, getByText } = makeSut({ healthy: false });

    const TITLE = getByText(
      'Você saiu da dieta dessa vez, mas continue se esforçando e não desista!'
    );

    expect(TITLE).toBeVisible();

    expect(toJSON).toMatchSnapshot();
  });
  it('Should be Image render correctly when healthy is false', () => {
    const { toJSON, getByTestId } = makeSut({ healthy: false });

    const IMAGE = getByTestId('image');

    expect(IMAGE).toHaveProp('source', 2);

    expect(toJSON).toMatchSnapshot();
  });
  it('Should be Image render correctly when healthy is true', () => {
    const { toJSON, getByTestId } = makeSut({ healthy: true });

    const IMAGE = getByTestId('image');

    expect(IMAGE).toHaveProp('source', 1);

    expect(toJSON).toMatchSnapshot();
  });
  it('Should be ConfirmRegister button render correctly', () => {
    const { toJSON, getByText } = makeSut({ healthy: true });

    const BUTTON = getByText('Ir para a tela inicial');

    expect(BUTTON).toBeVisible();

    expect(toJSON).toMatchSnapshot();
  });

  it('Should handleNavigateHome is called', () => {
    const { toJSON, getByText } = makeSut({ healthy: true });

    const BUTTON = getByText('Ir para a tela inicial');
    fireEvent.press(BUTTON);

    expect(mockHandleNavigateHome).toHaveBeenCalledTimes(1);

    expect(toJSON).toMatchSnapshot();
  });
});
