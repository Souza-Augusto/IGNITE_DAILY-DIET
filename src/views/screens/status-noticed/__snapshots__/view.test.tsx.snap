// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen:status-noticed view-model Should be ConfirmRegister button render correctly 1`] = `[Function]`;

exports[`Screen:status-noticed view-model Should be Image render correctly when healthy is false 1`] = `[Function]`;

exports[`Screen:status-noticed view-model Should be Image render correctly when healthy is true 1`] = `[Function]`;

exports[`Screen:status-noticed view-model Should be Noticed render correctly when healthy is false 1`] = `[Function]`;

exports[`Screen:status-noticed view-model Should be Noticed render correctly when healthy is true 1`] = `[Function]`;

exports[`Screen:status-noticed view-model Should be render correctly 1`] = `[Function]`;

exports[`Screen:status-noticed view-model Should be title render correctly when healthy is false 1`] = `[Function]`;

exports[`Screen:status-noticed view-model Should be title render correctly when healthy is true 1`] = `[Function]`;

exports[`Screen:status-noticed view-model Should handleNavigateHome is called 1`] = `[Function]`;

exports[`Screen:status-noticed view-model Should render correctly 1`] = `[Function]`;
